javascript/javascript1/week3/HW3.js

```diff
@@ -0,0 +1,211 @@
+    let day = (hoursToDay + durationOfSeries[i].days) / 365 / 80;
+    let timePercentage = day * 100;
+
+    totalTimeSpent += timePercentage;
+
+    console.log(
+      durationOfSeries[i].title +
+        " took " +
+        timePercentage.toFixed(2) +
+        " % of my life"
+    );
+  }
+  console.log(
+    "In total that is " + totalTimeSpent.toFixed(2) + "  % of my life"
+  );
+  return durationOfSeries;
+}
+
+const totalLifeSpan = seriesDurationOfLifespan(seriesDurations);
+//console.log(totalLifeSpan);
+
+//#4 My favorite songs
+
+const songDatabase = [
+  {
+    songId: 1,
+    title: "My baby",
+    artist: "Soggy socks"
+  },
+  {
+    songId: 2,
+    title: "3 nails in wood",
+    artist: "The carpenters"
+  },
+  {
+    songId: 3,
+    title: "Blacker than black",
+    artist: "Instant coffee"
+  },
+  {
+    songId: 4,
+    title: "When is enough too little?",
+    artist: "The spies girls"
+  }
+];
+
+const myPlaylist = [];
+
+//Add song to database
+
+let favSong1 = { songId: 5, title: "Shekere", artist: "Yemi Alade" };
+let favSong2 = { songId: 6, title: "ca bouge pas", artist: "Fally Ipupa" };
+let favSong3 = { songId: 7, title: "50 Naira", artist: "Sugar Boy" };
+
+function addSongToDatabase(song) {
+  songDatabase.push(song);
+  return songDatabase;
+}
+addSongToDatabase(favSong1);
+addSongToDatabase(favSong2);
+addSongToDatabase(favSong3);
+console.log(songDatabase);
+
+//Searching for a song
+
+const searchSong = [];
+function getSongByTitle(title) {
+  for (let i = 0; i < songDatabase.length; i++) {
+    if (title !== songDatabase[i].title) {
+      console.log("song not in database");
+    }
+    if (title === songDatabase[i].title) {
+      searchSong.push(songDatabase[i]);
+    }
+  }
+}
+
+getSongByTitle("Shekere");
+getSongByTitle("ca bouge pas");
+getSongByTitle("50 Naira");
+getSongByTitle("Wait");
+console.log(searchSong);
+
+//Create our own playlist
+function addSongToMyPlaylist(title) {
+  //let myPlaylist = [];
+  for (let i = 0; i < searchSong.length; i++) {
+    if (title === searchSong[i].title) {
+      myPlaylist.push(searchSong[i]);
+    }
+  }
+  return myPlaylist;
+}
+addSongToMyPlaylist("Shekere");
+console.log(myPlaylist);
+
+//#Step-3 NOnoNOnoYes (note taking app)
+
+const notes = [];
+function addnote(content, id) {
+  notes.push({ content, id });
+}
+addnote("Page", 11);
+addnote("Page", 4);
+console.log(notes);
+// Getting a note
+function getNoteFromId(id) {
+  for (let i = 0; i < notes.length; i++) {
+    if (id === notes[i].id) {
+      return notes[i];
+    } else if (id !== Number && !id) {
+      return "Error, id should begin as number and not  as an empty field";
+    } else {
+      return "oh no! wrong page";
+    }
+  }
+}
+console.log(getNoteFromId(11));
+//getting all notes
+console.log(notes);
+// Logging out notes
+function logOutNotesFormatted() {
+  for (let i = 0; i < notes.length; i++) {
+    console.log(
```

Comment:
Yes this works, but it's cleaner with template literals: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals